features/guidance/staggered-intersections.feature

```diff
@@ -98,3 +98,66 @@ Feature: Staggered Intersections
             | waypoints | route                         | turns                              |
             | a,g       | Oak St,Cedar Dr,Elm St,Elm St | depart,turn right,turn left,arrive |
             | g,a       | Elm St,Cedar Dr,Oak St,Oak St | depart,turn right,turn left,arrive |
+
+    Scenario: Staggered Intersection: do not collapse if a mode change is involved
+        Given the node map
+            """
+                j
+            a b c
+                d
+                e  f g
```

Comment:
These cases capture my fear very well. I would appreciate if you could also do a final one where only one of the sides is `pushing`, the segment in the middle and at the end are `driving`.

My reasoning here is that [here](https://github.com/Project-OSRM/osrm-backend/pull/3324/files#diff-99adcb8bccf8e94a176bdf92509ff070L819), we only have access to two of the contributing modes. We are missing the mode of the segment entering the staggered intersection. In a staggered intersection, three modes are part of the process. The segment coming in (currently untested in our check for a staggered intersection), the middle segment (which is the first of the two turns) and the out segment (the second parameter).

It seems that from the surrounding code, these might actually be considered correctly. But if we ever change the site of the call, we would have an untested regression.